src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,87 @@
-export const Pagination = () => {};
+import { getNumbers, getNumbersInRange } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handlePageChange = (page: number) => {
+    if (page >= 1 && page <= totalPages) {
+      onPageChange(page);
+    }
+  };
+
+  const firstItemOnPage = (currentPage - 1) * perPage + 1;
+  const lastItemOnPage = currentPage * perPage;
+
+  const pageItems = getNumbersInRange(
+    firstItemOnPage,
+    lastItemOnPage,
+    1,
+    total,
+  );
+
+  return (
+    <>
+      {/* Pagination links */}
+      <ul className="pagination">
+        {/* Previous page link */}
+        <li className={`page-item ${currentPage === 1 ? 'disabled' : ''}`}>
```

Comment:
it is better to use libraries like classnames or clsx for conditional classNames